docs/design/backups.md

```diff
@@ -0,0 +1,83 @@
+# Backup & Restore
+
+In particularly disastrous situations, we would like to be able to revert the etcd cluster to a previously known healthy state.
+This document lays out the design of how we want the healthy state to be captured, securely stored & safely restored to operator-managed etcd clusters.
+
+## Backup Resource
+
+We will introduce a new custom resource: `EtcdBackup`.
+
+```yaml
+apiVersion: etcd.improbable.io/v1alpha1
+kind: EtcdBackup
+metadata:
+  name: my-cluster-backup
+  namespace: default
+spec:
+  type: snapshot
+  destination:
+    gcsBucket:
+      bucketName: my-cluster-backup-bucket
+      creds:
+        secretRef: 
+          name: my-cluster-backup-bucket-credentials
+          namespace: default
+```
+
+This resource specifies that we desire a backup to be immediately taken as a [snapshot](https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/recovery.md#snapshotting-the-keyspace) from the etcd API.
+This snapshot will be saved to a GCS bucket, accessed with credentials read from a given secret.
```

Comment:
Is there anything to note here about the structure of the backup directory in GCS, aside from with a timestamp?

Reply:
no - i was imagining we would be something like

```
my-bucket
├── 2019-10-28T12:00:00Z.db
├── 2019-10-29T12:00:00Z.db
├── 2019-10-30T12:00:00Z.db
├── ...
```

we could make change the `bucketName` to hold the full path - something like 

```
location: gs://my-bucket/backups/2019-10-28T12:00:00Z.db
```

but this gets a bit weird with the schedule resource, we'd need some kind of name template in there to get unique names for each backup.

Reply:
will leave it as is. we can always introduce a `directory` field inside the resource - that's not a breaking change.